server/src/services/metadata.service.ts

```diff
@@ -589,9 +589,13 @@ export class MetadataService extends BaseService {
     let dateTimeOriginal = dateTime?.toDate();
     let localDateTime = dateTime?.toDateTime().setZone('UTC', { keepLocalTime: true }).toJSDate();
     if (!localDateTime || !dateTimeOriginal) {
-      this.logger.warn(`Asset ${asset.id} has no valid date, falling back to asset.fileCreatedAt`);
-      dateTimeOriginal = asset.fileCreatedAt;
-      localDateTime = asset.fileCreatedAt;
+      this.logger.warn(
+        `No valid date found in exif tags from asset ${asset.id}, falling back to earliest timestamp between file creation and file modification`,
+      );
+      // eslint-disable-next-line unicorn/prefer-math-min-max
```

Comment:
Why not do `Math.min(asset.fileModifiedAt, asset.fileCreatedAt)`?

Reply:
Because it returns a number, but I can do `new Date(Math.min(asset.fileModifiedAt, asset.fileCreatedAt))` instead if you prefer

Reply:
I'd prefer that, thanks!

Reply:
Sadly, ts linter reports it 😆
<img width="904" alt="image" src="https://github.com/user-attachments/assets/0c4c64d5-f695-4fab-943e-7d279bee761d" />
<img width="822" alt="image" src="https://github.com/user-attachments/assets/bf483015-2cf9-4f84-9b9a-4dcdc9cbf5cf" />
You have to choose the pill: the red or the green one ? 😄

Reply:
`Date.valueOf()` or `Date.getTime()` also work.
```typescript
new Date(Math.min(asset.fileModifiedAt.valueOf(), asset.fileCreatedAt.valueOf()))
```

Reply:
I don't want to take either pill and instead just use luxon lmao. Do you think that's overkill or would it make things nicely readable?

Reply:
Funny enough luxon's `min`/`max` functions require a `DateTime` instead of a `Date`, so a conversion is going to have to happen somewhere anyway, which means something like:
```typescript
DateTime.min(DateTime.fromJSDate(asset.fileModifiedAt), DateTime.fromJSDate(asset.fileCreatedAt)).toJSDate()
```
😅

Reply:
I personally prefer a vanilla approach, especially for this kind of simple need: 

> `Date.valueOf()` or `Date.getTime()` also work.
> 
> ```ts
> new Date(Math.min(asset.fileModifiedAt.valueOf(), asset.fileCreatedAt.valueOf()))
> ```

Which one you prefer? I'll commit with this one

Reply:
Your latest solution is great! Sorry haha

Reply:
great!